core/src/main/java/com/es/core/model/phone/JdbcPhoneDao.java

```diff
@@ -9,19 +8,42 @@
 import java.util.Optional;
 
 @Component
-public class JdbcPhoneDao implements PhoneDao{
+public class JdbcPhoneDao implements PhoneDao {
     @Resource
     private JdbcTemplate jdbcTemplate;
 
     public Optional<Phone> get(final Long key) {
-        throw new UnsupportedOperationException("TODO");
+        return jdbcTemplate.query("select ph.*, colors.code as color " +
+                        "from (select phones.* from phones where phones.id = ?) ph " +
+                        "left join phone2color on ph.id = phone2color.phoneId " +
+                        "left join colors on phone2color.colorId = colors.id",
+                new Object[]{key}, new PhoneRowMapper()).stream().findAny();
     }
 
     public void save(final Phone phone) {
-        throw new UnsupportedOperationException("TODO");
+        jdbcTemplate.update("INSERT INTO phones VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)",
```

Comment:
Looks not really cool, at least name values like "UPDATE phones SET price = ?, displaySizeInches = ? ......", current implementation is based on db columns direction, which can be usually change and your code will become hard to refactor